rct229/rulesets/ashrae9012019/section4/section4rule14.py

```diff
@@ -0,0 +1,103 @@
+from rct229.schema.config import ureg
+from rct229.schema.schema_enums import SchemaEnums
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.pint_utils import ZERO, CalcQ
+
+LIGHTING_SPACE = SchemaEnums["LightingSpaceOptions2019ASHRAE901TG37"]
+ENERGY_SOURCE = SchemaEnums["EnergySourceOptions"]
+
+REQ_EQUIP_POWER_DENSITY = 20 * ureg("W/ft2")
+
+
+class Section4Rule14(RuleDefinitionListIndexedBase):
+    """Rule 14 of ASHRAE 90.1-2019 Appendix G Section 4 (Schedule-Setpoints)"""
+
+    def __init__(self):
+        super(Section4Rule14, self).__init__(
+            rmrs_used=produce_ruleset_model_instance(
+                USER=False, BASELINE_0=False, PROPOSED=True
+            ),
+            each_rule=Section4Rule14.SpaceRule(),
+            index_rmr=PROPOSED,
+            id="4-14",
+            description="A computer room is defined as a room whose primary function is to house equipment for the processing and storage of electronic data and that has a design electronic data equipment power density exceeding 20 W/ft2 of conditioned floor area.",
+            ruleset_section_title="Schedule - Setpoints",
+            standard_section="Section 3 Definitions",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_descriptions/0",
+            list_path="$.buildings[*].building_segments[*].zones[*].spaces[*]",
+        )
+
+    def create_data(self, context, data):
+        rmd_p = context.proposed
+
+        sch_ids_p = find_all(
+            f'$.buildings[*].building_segments[*].zones[*].spaces[*].miscellaneous_equipment[*][?(@.energy_type="ELECTRICITY")].multiplier_schedule',
+            rmd_p,
+        )
+        schedule_p = []
+        for scd_id_p in sch_ids_p:
+            schedule_p += find_all(f'$.schedules[*][?(@id = "{scd_id_p}")]', rmd_p)
+
+        return {"schedule_p": schedule_p}
+
+    class SpaceRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section4Rule14.SpaceRule, self).__init__(
+                rmrs_used=produce_ruleset_model_instance(
+                    USER=False, BASELINE_0=False, PROPOSED=True
+                ),
+                required_fields={
+                    "$": [
+                        "lighting_space_type",
+                        "miscellaneous_equipment",
+                        "floor_area",
+                    ],
+                },
+                fail_msg="The space has been classed as a computer room in terms of the lighting space type but the electronic data equipment power density does not appear to exceed 20 w/sf.",
+            )
+
+        def get_calc_vals(self, context, data=None):
+            space_p = context.proposed
+
+            schedule_p = data["schedule_p"]
+            floor_area_p = space_p["floor_area"]
+
+            total_space_misc_Wattage_including_multiplier_p = ZERO.POWER
+            if space_p["lighting_space_type"] == LIGHTING_SPACE.COMPUTER_ROOM:
```

Comment:
follow the above comment, we do not need to do a filter here. Any space reached here shall be guaranteed as a computer room.